server/service/src/invoice_line/stock_out_line/set_prescribed_quantity/mod.rs

```diff
@@ -0,0 +1,338 @@
+        );
+    }
+
+    #[actix_rt::test]
+    async fn set_prescribed_quantity_existing_stock_line() {
+        let (_, _, connection_manager, _) = setup_all(
+            "set_prescribed_quantity_existing_stock_line",
+            MockDataInserts::all(),
+        )
+        .await;
+
+        let service_provider = ServiceProvider::new(connection_manager);
+        let context = service_provider
+            .context(mock_store_a().id, "".to_string())
+            .unwrap();
+        let service = service_provider.invoice_line_service;
+
+        let result = service.set_prescribed_quantity(
+            &context,
+            SetPrescribedQuantity {
+                invoice_id: mock_prescription_a_invoice_line_a().invoice_id,
+                item_id: mock_prescription_a_invoice_line_a().item_link_id,
+                prescribed_quantity: 10.0,
+            },
+        );
+
+        assert!(result.is_ok());
+
+        let invoice_line = result.unwrap();
+
+        assert_eq!(
+            invoice_line.invoice_line_row.invoice_id,
+            mock_prescription_a_invoice_line_a().invoice_id
+        );
+
+        assert_eq!(
+            invoice_line.invoice_line_row.item_link_id,
+            mock_prescription_a_invoice_line_a().item_link_id
+        );
+
+        assert_eq!(
+            invoice_line.invoice_line_row.item_code,
+            mock_prescription_a_invoice_line_a().item_code
+        );
+
+        assert_eq!(
+            invoice_line.invoice_line_row.item_name,
+            mock_prescription_a_invoice_line_a().item_name
+        );
+
+        assert_eq!(
+            invoice_line.invoice_line_row.prescribed_quantity,
+            Some(10.0)
+        );
+    }
+
+    #[actix_rt::test]
+    async fn set_prescribed_quantity_multiple_lines_with_unallocated() {
+        let (_, _, connection_manager, _) = setup_all_with_data(
+            "set_prescribed_quantity_multiple_lines_with_unallocated",
+            MockDataInserts::all(),
+            MockData {
+                invoice_lines: vec![
+                    mark_prescription_unallocated_invoice_line(),
+                    mock_existing_stock_invoice_line(),
+                ],
+                ..Default::default()
+            },
+        )
+        .await;
+
+        let service_provider = ServiceProvider::new(connection_manager);
+        let context = service_provider
+            .context(mock_store_a().id, "".to_string())
+            .unwrap();
+        let service = service_provider.invoice_line_service;
+
+        let result = service.set_prescribed_quantity(
+            &context,
+            SetPrescribedQuantity {
+                invoice_id: mark_prescription_unallocated_invoice_line().invoice_id,
+                item_id: mark_prescription_unallocated_invoice_line().item_link_id,
+                prescribed_quantity: 10.0,
+            },
+        );
+
+        assert!(result.is_ok());
+
+        let invoice_line = result.unwrap();
+
```

Comment:
I think you need a test here to see how many lines exist after this function runs. 
We should be deleting the unallocated line right?

Reply:
Would also like a test where there's more than one allocated stockline for the item.
We should check that there's only one line with prescribed quantity set on it in this scenario.